core/src/apps/cardano/sign_tx.py

```diff
@@ -340,8 +329,11 @@ def _cborize_tx_body(keychain: seed.Keychain, msg: CardanoSignTx) -> dict:
 
     # tx_body[6] is for protocol updates, which we don't support
 
-    if msg.metadata:
-        tx_body[7] = _hash_metadata(bytes(msg.metadata))
+    if msg.auxiliary_data:
+        auxiliary_data_cbor = get_auxiliary_data_cbor(
+            keychain, msg.auxiliary_data, msg.protocol_magic, msg.network_id
+        )
+        tx_body[7] = hash_auxiliary_data(bytes(auxiliary_data_cbor))
```

Comment:
Sadly no, since protobuf `bytes` are actually received as `bytearray` both from trezorlib and Trezor Connect. I can raise it with Trezor guys to find out more.